### PR TITLE
fix(markdown): decode assets link to ensure bundler can find the file

### PR DESCRIPTION
**Pull request by Mister-Hope**
This is an issue in all VuePress versions.

```md
![中国](./中国.png)
```

should be allowed. But `markdown-it` 's default behavior will encode it to `./%E4%B8%AD%E5%9B%BD.png`, so that  bundler won't be able to find the image file.

I made a fix in V1 by my `vuepress-plugin-md-enhance`, but I think V2 can support it.

### CODE REVIEW

File: packages/@vuepress/markdown/src/plugins/assetsPlugin.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import type { PluginWithOptions } from 'markdown-it'
+import { decode } from 'mdurl'
```
Comment (meteorlxy) [inline packages/@vuepress/markdown/src/plugins/assetsPlugin.ts:2]: `mdurl` is not listed in the `dependencies` of markdown package.

We should add it explicitly.

**Review by meteorlxy — CHANGES_REQUESTED** [review]

File: packages/@vuepress/markdown/src/plugins/assetsPlugin.ts (hunk 1 above)
Comment (Mister-Hope) [reply packages/@vuepress/markdown/src/plugins/assetsPlugin.ts:2]: Got it, I forgot adding because it's already on the markdown-it toolchain.

**Comment by meteorlxy** [comment]
I just added more test cases, and I think we should make all those cases passed

**Comment by Mister-Hope** [comment]
I don't think alias is needed, as I am not thinking of any situation which a person should define a alia containing chinese. While a markdown folder likely to contain chinese images,(like harmony os official docs)

Even if we resolve those alias begin with @, alias are not forced to begin with `@`.

The only way to fit this is to decode back every link not begin with `/` , comparing with decoding every relative links now.

**Comment by meteorlxy** [comment]
Maybe just decode the original `src` is enough. No need to add to many codes

**Review by meteorlxy — CHANGES_REQUESTED** [review]

**Review by meteorlxy — APPROVED** [review]
Made some refactor. Thanks for your contribution :heart:

**Pull request merged**